src/impl.cpp

```diff
@@ -0,0 +1,25 @@
+#include "impl.h"
+
+char* IntToBinaryRepresentation(char *buff,int value)
+{
+   constexpr unsigned int int_size = sizeof(int) * 8;
+   for (unsigned int i = 0; i < int_size; i++)
+   {
+       buff[i] = '0' + static_cast<char>((value >> (int_size - 1 - i)) & 0x1);
+   }
+   buff[int_size] = '\0';
+
+   return buff;
+}
+
+int BinaryInvert(int x,int p,int n)
+{
+    unsigned int mask = x;
```

Comment:
**why do you assign the x to mask? what if x will be 0?
then xor of your mask and the number will give 0 value, so it would not be inverted**